driver-core/src/main/java/com/datastax/driver/core/Session.java

```diff
@@ -41,6 +42,19 @@
  */
 public interface Session extends Closeable {
 
+  /**
+   * Sets desired factory for tracing information for this Session. By default it is {@link
+   * com.datastax.driver.core.tracing.NoopTracingInfoFactory}
+   */
+  void setTracingInfoFactory(TracingInfoFactory tracingInfoFactory);
```

Comment:
Small nit: a fully complete JavaDoc would also have: `@param tracingInfoFactory description of this parameter`.